client/components/web-preview/content.jsx

```diff
@@ -406,21 +407,63 @@ export default class WebPreviewContent extends Component {
 					) }
 					{ 'seo' !== this.state.device && (
 						<div
+							onMouseEnter={ () => {
+								if ( this.props.isLaunchpad ) {
+									this.setState( { showIFrameOverlay: true } );
+								}
+							} }
+							onMouseLeave={ () => {
+								if ( this.props.isLaunchpad ) {
+									this.setState( { showIFrameOverlay: false } );
+								}
+							} }
 							className={ classNames( 'web-preview__frame-wrapper', {
 								'is-resizable': ! this.props.isModalWindow,
 							} ) }
 						>
 							<iframe
 								ref={ this.setIframeInstance }
 								className="web-preview__frame"
-								style={ { ...this.state.iframeStyle, height: this.state.viewport?.height } }
+								style={ {
+									...this.state.iframeStyle,
+									height: this.state.viewport?.height,
+									pointerEvents: this.props.isLaunchpad ? 'auto' : 'all',
+								} }
 								src="about:blank"
 								onLoad={ () => this.setLoaded( 'iframe-onload' ) }
 								title={ this.props.iframeTitle || translate( 'Preview' ) }
 								fetchpriority={ fetchpriority ? fetchpriority : undefined }
 								scrolling={ autoHeight ? 'no' : undefined }
 								tabIndex={ disableTabbing ? -1 : 0 }
 							/>
+							{ this.props.isLaunchpad && (
+								<div
+									className="web-preview__frame-edit-overlay"
+									style={ {
+										opacity: this.state.showIFrameOverlay ? '1' : '0',
+										background: this.state.showIFrameOverlay
+											? `rgba(16, 21, 23, 0.5)`
+											: 'rgba(16, 21, 23, 0)',
+										transition: 'background 0.2s ease',
+									} }
+								>
+									<button
+										style={ {
+											position: 'relative',
+											top: this.state.showIFrameOverlay ? '0' : '15px',
+											transition: 'all 0.2s ease',
+										} }
+										className="web-preview__frame-edit-button"
+										onClick={ () => {
+											window.location.assign( `/site-editor/${ this.props.externalUrl }` );
+										} }
+										onFocus={ () => this.setState( { showIFrameOverlay: true } ) }
+										onBlur={ () => this.setState( { showIFrameOverlay: false } ) }
+									>
+										{ translate( 'Edit' ) }
```

Comment:
Currently, my mac screen reader only dictates this button as the "Edit" button which might not be the best experience for sight challenged users.

Maybe we could consider changing the text of the button itself to "Edit Site"? Otherwise, we could add something like `aria-label="Edit your new site"`

Reply:
For reference https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA8